solr/core/src/java/org/apache/solr/handler/component/SearchHandler.java

```diff
@@ -489,6 +491,10 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
             .getResponseHeader()
             .asShallowMap()
             .put(SolrQueryResponse.RESPONSE_HEADER_PARTIAL_RESULTS_KEY, Boolean.TRUE);
+      } catch (IndexSearcher.TooManyClauses | TooManyBasicQueries ex) {
+        // An IOException on a non-distributed request is typically an issue parsing the query at
+        // the lucene level
+        throw new SolrException(SolrException.ErrorCode.BAD_REQUEST, ex);
```

Comment:
In the slack thread discussing this issue I think Cam had suggested a more specific 4xx error. Not sure if it's worth adding a constant for this code? 

422 Unprocessable Entity
The request was well-formed but was unable to be followed due to semantic errors.[[9]](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#cite_note-RFC_4918-10)

Reply:
Fundamentally I agree that these should probably be 422 instead of 400 as a more specific error, but the Solr code and our plugin is using the BAD_REQUEST (400) for all query parsing issues currently so I think it'll make more sense to be consistent here and 400 should still be recognized as a response that cannot be retried.

Reply:
makes sense!